src/lint.ts

```diff
@@ -28,6 +45,47 @@ export const lintFile = async (filePath: string) => {
   return [...fileDiagnostics, ...textDiagnostics]
 }
 
+/**
+ * Analyses and produces a set of diagnostics for the folder at the given path.
+ * @param {string} folderPath - the path to the folder to be linted.
+ * @param {LintConfig} configuration - an optional configuration. When not passed in, this is read from the .sasjslint file.
+ * @returns {Diagnostic[]} array of diagnostic objects, each containing a warning, line number and column number.
+ */
+export const lintFolder = async (
+  folderPath: string,
+  configuration?: LintConfig
+) => {
+  const config = configuration || (await getLintConfig())
+  const diagnostics: Diagnostic[] = []
+  const fileNames = await listSasFiles(folderPath)
+  await asyncForEach(fileNames, async (fileName) => {
+    diagnostics.push(
+      ...(await lintFile(path.join(folderPath, fileName), config))
+    )
+  })
+
+  const subFolders = (await listSubFoldersInFolder(folderPath)).filter(
+    (f: string) => !excludeFolders.includes(f)
+  )
+
+  await asyncForEach(subFolders, async (subFolder) => {
+    diagnostics.push(
+      ...(await lintFolder(path.join(folderPath, subFolder), config))
+    )
+  })
+
+  return diagnostics
+}
+
+/**
+ * Analyses and produces a set of diagnostics for the current project.
+ * @returns {Diagnostic[]} array of diagnostic objects, each containing a warning, line number and column number.
+ */
+export const lintProject = async () => {
+  const projectRoot = await getProjectRoot()
```

Comment:
Handle case in which, `getProjectRoot()` returns `''`. Means there's no .lint config found.

For cli, we are using
```
const lintConfigFolder =
    (await getDirectoryContainingLintConfig()) ||
    process.projectDir ||
    process.currentDir
```

Reply:
Done!